src/include/misc.i

```diff
@@ -193,15 +193,44 @@ __wt_snprintf_len_incr(
  *	Complain if a transaction is/isn't running.
  */
 static inline int
-__wt_txn_context_check(WT_SESSION_IMPL *session, bool requires_txn)
+__wt_txn_context_check(
+    WT_SESSION_IMPL *session, bool requires_txn, bool prepare_allowed)
 {
+	if (!requires_txn && F_ISSET(&session->txn, WT_TXN_RUNNING))
+		WT_RET_MSG(session, EINVAL,
+		    "%s: not permitted in a running transaction",
+		    session->name);
 	if (requires_txn && !F_ISSET(&session->txn, WT_TXN_RUNNING))
 		WT_RET_MSG(session, EINVAL,
 		    "%s: only permitted in a running transaction",
 		    session->name);
-	if (!requires_txn && F_ISSET(&session->txn, WT_TXN_RUNNING))
+#ifdef HAVE_TIMESTAMPS
```

Comment:
Why didn't you call `__wt_txn_context_prepare_check` rather than duplicating the code?